Source/ARTRestChannel.m

```diff
@@ -305,19 +306,37 @@ - (void)internalPostMessages:(id)data callback:(void (^)(ARTErrorInfo *__art_nul
             return;
         }
     }
-    
-    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:[self->_basePath stringByAppendingPathComponent:@"messages"]]];
+
+    NSString *randomId = [NSUUID new].UUIDString;
```

Comment:
Could you avoid generating here, only assigning `randomId` within the check for `options.addRequestIds` below? For those that don't want request ids, which for now will be most users, this may have a slight performance hit. Also, less 'new code' run for every customer upgrading their library.

Reply:
Ups, I forgot to remove that. Fixed 2f87540.